numpy/core/src/multiarray/descriptor.c

```diff
@@ -1301,6 +1301,12 @@ _convert_from_dict(PyObject *obj, int align)
             goto fail;
         }
     }
+
+    Py_XDECREF(fields);
+    Py_XDECREF(names);
+    Py_XDECREF(descrs);
+    Py_XDECREF(offsets);
+    Py_XDECREF(titles);
```

Comment:
/home/sebastian/forks/numpy/numpy/core/src/multiarray/descriptor.c

Reply:
`names` is always null here, it's set on line 1235

Reply:
As for `offsets`, `descrs`, and `titles`, I'd be inclined to decref and null `descrs`, `offsets`, and `titles` on 1212, where they are last used

Reply:
Hmm, maybe better. The nicer pattern would be to use the same cleanup code for success and failure path I think, which is why this unnecessary decref got in. But yeah, I can just decref it all there and think about moving things some other day

Reply:
I suppose I could see it either way. I'll let you make the call.